wordpress-mobile/WordPress-Android
# Add dismiss tracking to system notifications

Fixes #10626

In this PR I'm adding tracking of all the notification dismisses I could find. 
As the first step I've extracted `NotificationPushIds` that were scattered all over the place into a separate `object`. I've created a `NotificationType` enum which is used to track the type of dismissed notification. Not all the notification types might be super clear but it's not obvious what's happening when they are shown (and some are covering more cases). I tried to be as granular as possible and follow the types in the code. The code is really hard to read and follow. 

To test:
* Have the app show you a notification (for example when a blog you follow publishes a post or there is a comment on any of your posts)
* Dismiss the notification
* Check that the notification is tracked in `LogCat` 
* Check that the behaviour of the notifications hasn't changed

PR submission checklist:
